### DISCUSSION TITLE
Difference between subscription and stream in flutter

### DISCUSSION THREAD

**Discussion by hhamja** [supabase/supabase#8617] [Questions] [answered]
I now don't understand the difference between subscription and stream in flutter.

Should I use a subscription and a stream together, or if I use a stream, I don't need to use a subscription?

**Comment by dshukertjr** [comment, marked as answer]
`.stream()` is an easy way to listen to data on the database including the initial data and streams it as `Stream` type. `.on().subscribe()` is a method to listen to database changes using callbacks. It is what is used within `.stream()` to power the realtime updates. `.on().subscribe()` also has additional features such as [presence and broadcast](https://supabase.com/blog/supabase-realtime-with-multiplayer-features). 

In most cases, using `.stream()` should do it as far as getting realtime updates from the database!
